eth2near/eth2near-block-relay/index.js

```diff
@@ -184,35 +196,42 @@ class Eth2NearRelay {
           for (let i = clientBlockNumber + 1; i <= endBlock; i++) {
             if (this.validateHeaderMode === 'bsc') {
               const block = await this.robustWeb3.getBlock(i)
+              if (await isBlockAlreadySubmitted(block)) {
+                if (endBlock <= chainBlockNumber) {
+                  endBlock++
+                  continue
+                }
+              }
               blockPromises.push({ header_rlp: block })
             } else {
               blockPromises.push(this.getParseBlock(i))
             }
           }
 
           const blocks = await Promise.all(blockPromises)
+          const firstBlock = blocks[0].header_rlp
           console.log(
-            `Got and parsed block ${clientBlockNumber + 1} to block ${endBlock}`
+            `Got and parsed block ${firstBlock.number} to block ${endBlock}`
```

Comment:
As you are doing some little refactoring here already, let's eliminate a possible confusion so `endBlock` should be `endBlockNumber`

Reply:
Fixed 147a517f09f3adeada85f845260a40d60b2ba346